python/cuml/utils/input_utils.py

```diff
@@ -201,7 +201,13 @@ def convert_dtype(X, to_dtype=np.float32):
                                 "in data loss.")
             return X_m
 
+    elif isinstance(X, cudf.Series):
+        return X.astype(to_dtype)
+
     elif cuda.is_cuda_array(X):
+        if hasattr(X, 'dtype') and np.dtype(X.dtype) == to_dtype or \
+                hasattr(X, 'typestr') and np.dtype(X.typestr) == to_dtype:
+            return X
```

Comment:
Adding to my prior review (sorry for jumping the gun so fast...) I don't understand the need for this if condition here, particularly now that cupy is required, and this return seems to be what is causing the error in CI.